src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1674,18 +1674,28 @@ void TorrentImpl::handleTrackerWarningAlert(const lt::tracker_warning_alert *p)
 
 void TorrentImpl::handleTrackerErrorAlert(const lt::tracker_error_alert *p)
 {
-    const QString trackerUrl = p->tracker_url();
-
     // Starting with libtorrent 1.2.x each tracker has multiple local endpoints from which
     // an announce is attempted. Some endpoints might succeed while others might fail.
     // Emit the signal only if all endpoints have failed.
-    const QVector<TrackerEntry> trackerList = trackers();
-    const auto iter = std::find_if(trackerList.cbegin(), trackerList.cend(), [&trackerUrl](const TrackerEntry &entry)
+    const std::vector<lt::announce_entry> trackerList = m_nativeHandle.trackers();
```

Comment:
@glassez this call now takes ~33% of main thread time during startup, any way to avoid this?

Reply:
>@glassez this call now takes ~33% of main thread time during startup, any way to avoid this?

AFAIK, this is only way to obtain torrent trackers from libtorrent side so we can't avoid it.

Reply:
My release builds got mixed up, this only takes ~5%